### PR TITLE
Revert "Revert "When we pull in a cacheline, we should use it too.""

### PR DESCRIPTION
**Pull request by baldersheim**
Reverts vespa-engine/vespa#13499

### CODE REVIEW

**Comment by baldersheim** [comment]
Depends on #13515

**Comment by baldersheim** [comment]
This is a pure reapply. The issue causing the revert is addressed in #13515. It was triggered by having alignment above the standard which is 16.

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp [hunk 1]
```diff
@@ -1,61 +1,91 @@
...
-            _lastMaxDocIdLimit = (index + 1) * WordLen;
-        } else {
+        if (__builtin_expect(docId >= _numDocs, false)) {
             setAtEnd();
+            return;
+        }
+        const uint32_t index(wordNum(docId));
+        if (docId >= _lastMaxDocIdLimitRequireFetch) {
+            uint32_t baseIndex = index & ~(sizeof(_lastWords)/sizeof(Word) - 1);
+            _update(_accel, baseIndex, _bvs, _lastWords);
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp:84]: I think the magic should be done outside the update call and the update call just forward to the appropriate function.

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp [hunk 2]
```diff
@@ -1,61 +1,91 @@
...
+        : MultiBitVectorIterator<Update>(std::move(children))
+    { }
 private:
     void doSeek(uint32_t docId) override { this->strictSeek(docId); }
     Trinary is_strict() const override { return Trinary::True; }
 };
 
+struct And {
+    using Word = BitWord::Word;
+    void operator () (const IAccelrated & accel, size_t offset, const std::vector<std::pair<const void *, bool>> & src, Word *dest) {
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp:59]: offset should already be bytes here, and destination void.

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp [hunk 3]
```diff
@@ -1,61 +1,91 @@
...
+        accel.and64(offset*sizeof(uint64_t), src, dest);
```
Comment (havardpe) [inline searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp:60]: offset is probably in Words and not in uint64_t, but I think this magic should be done together with the other magic anyways.

Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp]: Fixed

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp (hunk 2 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp]: Fixed

File: searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp (hunk 1 above)
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/queryeval/multibitvectoriterator.cpp]: Fixed

**Pull request merged**